### DISCUSSION TITLE
chezmoiexternal vs install scripts

### DISCUSSION THREAD

**Discussion by MrMarble** [twpayne/chezmoi#2197] [Q&A] [answered]
Hi! I've been using chezmoi for about a week, and I'm loving it so far! I'm almost done migrating all my config to chezmoi, but I have a question.

What is the correct/intended way of managing external dependencies? I'm pushing all the way using the chezmoiexternal file, but looking through some `dotfiles` repositories (including yours @twpayne 👀) I'm seeing a lot of people using scripts to install part of the external tools they use.

Is there any reasoning for that usage or just personal preference?

**Comment by twpayne** [comment, marked as answer]
`.chezmoiexternal` is for pulling files, directories, and git repos into your home directory.

Scripts can be used for anything, e.g. using a package manager to install packages you want.

I would recommend using `.chezmoiexternal` if possible, and a script only if necessary.
